std/zip.d

```diff
@@ -1137,59 +1131,117 @@ public:
+            _segs = [Segment(0, data.length)];
+            auto eocdr = getRecord!EndOfCentralDirRecord(start);
+            if (eocdr.signature != endOfCentralDirSignature) continue;
 
-            auto numberOfStartOfCentralDirectory = getUshort(start + 6);
-            if (numberOfStartOfCentralDirectory != 0) continue; // dito
+            if (fromLE!ushort(eocdr.numberOfThisDisk) != 0) continue; // no support for multiple volumes yet
+            if (fromLE!ushort(eocdr.numberOfCDDisk) != 0) continue; // dito
 
-            if (numberOfThisDisc < numberOfStartOfCentralDirectory) continue;
+            ulong k = start - Zip64EndOfCentralDirLocatorRecord.sizeof;
+            auto maybeZip64 = k < start && data[k .. k + 4] == zip64EndOfCentralDirLocatorSignature;
 
-            uint k = start - zip64EndOfCentralDirLocatorLength;
-            auto maybeZip64 = k < start && _data[k .. k + 4] == zip64EndOfCentralDirLocatorSignature;
-
-            auto totalNumberOfEntriesOnThisDisk = getUshort(start + 8);
-            auto totalNumberOfEntriesInCentralDir = getUshort(start + 10);
+            auto totalNumberOfEntriesOnThisDisk = fromLE!ushort(eocdr.entriesOnThisDisk);
+            auto totalNumberOfEntriesInCentralDir = fromLE!ushort(eocdr.entries);
 
             if (totalNumberOfEntriesOnThisDisk > totalNumberOfEntriesInCentralDir &&
                (!maybeZip64 || totalNumberOfEntriesOnThisDisk < 0xffff)) continue;
 
-            auto sizeOfCentralDirectory = getUint(start + 12);
+            auto sizeOfCentralDirectory = fromLE!uint(eocdr.size);
             if (sizeOfCentralDirectory > start &&
                (!maybeZip64 || sizeOfCentralDirectory < 0xffff)) continue;
 
-            auto offsetOfCentralDirectory = getUint(start + 16);
-            if (offsetOfCentralDirectory > start - sizeOfCentralDirectory &&
+            auto offsetOfCentralDirectory = fromLE!uint(eocdr.offset);
+            if (offsetOfCentralDirectory + sizeOfCentralDirectory > start &&
                (!maybeZip64 || offsetOfCentralDirectory < 0xffff)) continue;
 
-            auto zipfileCommentLength = getUshort(start + 20);
-            if (start + zipfileCommentLength + endOfCentralDirLength != data.length) continue;
-
-            enforce!ZipException(endrecOffset == to!uint(data.length),
-                                 "found more than one valid 'end of central dir record'");
+            if (start + EndOfCentralDirRecord.sizeof + fromLE!ushort(eocdr.commentLength) != data.length) continue;
 
+            enforce!ZipException(endrecOffset == data.length, "found more than one valid 'end of central dir record'");
             endrecOffset = start;
         }
 
-        enforce!ZipException(endrecOffset != to!uint(data.length),
+        enforce!ZipException(endrecOffset != data.length,
                              "found no valid 'end of central dir record'");
 
         return endrecOffset;
     }
 
+    private void checkForZip64(ulong offset)
```

Comment:
This is only used once so please make it a `private bool checkForZip64(ulong offset) const` and assign to `_isZip64` at the use site (assuming it at false by default.